docs/src/architecture/notes/note_design.md

```diff
@@ -0,0 +1,26 @@
+# Note design
+The diagram below illustrates the contents of a note:
+
+<p align="center">
+    <img src="../../diagrams/architecture/note/Note.png">
+</p>
+
+As shown in the above picture:
+* **Vault &rarr;** a set of assets that are stored in note
+* **Script &rarr;** it must be executed in a context of some account to claim the assets
+* **Inputs &rarr;** these are placed onto the stack before a note's script is executed
+* **Serial number &rarr;** a note's unique identifier
```

Comment:
Given the current content for each of these, I'm not sure they deserve their own sections. We could probably move most of the content into the bullet points and then have a few separate paragraphs addressing specific points we want to make.

Reply:
I added the core information to the bullet list and added two paragraphs explaining some specifics.